src/main/kotlin/io/zerobase/smarttracing/resources/OrganizationsResource.kt

```diff
@@ -1,47 +1,150 @@
+        val name = request.name
+        val phone = request.contact.phone
+        val email = request.contact.email
+        val contactName = request.contact.contactName
+        val address = request.address
+        val hasTestingFacilities = request.hasTestingFacilities ?: false
+        val hasMultipleSites = request.hasMultipleSites ?: true
+
+        val id = dao.createOrganization(
+            name, phone, email,
+            contactName, address,
+            hasTestingFacilities,
+            hasMultipleSites
+        )
+
+        return when (id) {
+            null -> null
+            else -> IdWrapper(id)
+        }
     }
 
     @Path("/{id}/sites")
     @POST
     @Creator
-    fun createSite(@PathParam("id") id: String, request: Any): IdWrapper {
-        TODO("not implemented")
+    fun createSite(@PathParam("id") id: String, request: CreateSiteRequest): IdWrapper? {
+        val name = request.name ?: ""
+        val category = request.category
+        val subcategory = request.subcategory
+        val latitude = request.location.latitude
+        val longitude = request.location.longitude
+        val isTesting = request.isTesting
+        val isCat = siteTypes.containsKey(category)
+        val isSubCat = siteTypes[category]?.contains(subcategory)
+
+        if (!(isCat!!)) {
+            return null
+        }
+
+        if (!(isSubCat!!)) {
+            return null
+        }
+
+        val id = if (request.contact == null) {
+            dao.createSite(
+                id, name, category, subcategory, latitude,
+                longitude, isTesting, null, null, null
+            )
+        } else {
+            dao.createSite(
+                id, name, category, subcategory, latitude,
+                longitude, isTesting, request.contact!!.phone,
+                request.contact!!.email, request.contact!!.contactName
+            )
+        }
+
+        return when (id) {
+            null -> null
+            else -> IdWrapper(id)
+        }
     }
 
     @Path("/{id}/sites")
     @GET
     fun getSites(@PathParam("id") id: String): List<Any> {
-        TODO("not implemented")
+        return dao.getSites(id)!!
     }
 
     @Path("/{id}/multiple-sites-setting")
     @PUT
     fun updateMultipleSitesSetting(@PathParam("id") id: String, hasMultipleSites: Boolean) {
-        TODO("not implemented")
+        dao.setMultiSite(id, hasMultipleSites)
     }
 
     @Path("/{orgId}/sites/{siteId}/scannables")
     @POST
     @Creator
-    fun createScannable(@PathParam("orgId") orgId: String, @PathParam("siteId") siteId: String, request: Any): IdWrapper {
-        TODO("not implemented")
+    fun createScannable(@PathParam("orgId") orgId: String, @PathParam("siteId") siteId: String, request: CreateScannableRequest): IdWrapper? {
+        val type = request.type
+        val singleUse = request.singleUse
+
+        if (!scanTypes.contains(type)) {
+            return null
+        }
```

Comment:
create a custom exception and an exception mapper instead of returning `null`

Reply:
why not use WebApplicationException?

Reply:
you can use that if you want, but limit it's use to the resource classes. in the dao, use custom exceptions and a mapper because the http framework has no business being in the database code. we probably need to work with the front-end to define a standard error structure that we can put as a body on error responses. in the meantime, just return a 400 for validation errors